v3/search.go

```diff
@@ -254,8 +263,16 @@ func (e *Entry) Unmarshal(i interface{}) (err error) {
 			}
 		case string:
 			fv.SetString(values[0])
+		case []byte:
+			fv.SetBytes([]byte(values[0]))
+		case int:
+			intVal, err := strconv.ParseInt(values[0], 10, 32)
+			if err != nil {
+				return fmt.Errorf("ldap: could not parse value '%s' into int field", values[0])
+			}
+			fv.SetInt(int64(intVal))
```

Comment:
Why this additional type conversion to int64 here? `strconv.ParseInt` already returns an int64 value